linen_examples/imagenet/imagenet_lib.py

```diff
@@ -210,10 +210,12 @@ def save_checkpoint(state, model_dir):
     step = int(state.step)
     checkpoints.save_checkpoint(model_dir, state, step, keep=3)
 
+def avg_across_axis(x, axis_name='x'):
```

Comment:
Why not pmap this function right away?

so

```
@functools.partial(jax.pmap, axis_name='x')
def avg_across_axis(x):
  return lax.pmean(x, 'x')
```

this way it's a bit more clear that the pmap will be compiled just once.
Otherwise we need to really on some smart caching behavior.